### ISSUE TITLE
Half speed over Ethernet

### ISSUE THREAD

**Issue by qy2009** [natecarlson/quectel-rgmii-configuration-notes#3]
Hi,

Love your notes, they are so helpful :) 
One thing I noticed is that if I use USB for the internet, I can get around 300Mbps download, and 100Mbps upload, however, once I changed to using RJ45 for the internet, I can only get around 150 Mbps - 200 Mbps download, upload is still around 100Mbps. 

Do you know why? 

So I have a WG1608 with 520, with the same location, that router I can only get around 200 Mbps download and 100 Mbps upload, however, my phone can easily get around 500 Mbps download, 10 Mbps upload. ( I have the 520 connected with an external antenna on my roof).

That's why I bought the M2 RJ45, and thinking if I could increase the speed and get rid of that poor 7610 CPU. But it turned out to be similar, or even a little bit worse. I noticed that other people have similar issues with WG1608 + 520 + Rooter firmware.

thanks!

**Comment by natecarlson** [comment]
Are you on a RM520 or a 502? If it's at a RM502Q-AE, you need the newest firmware release, it should take care of it. If it's a 520, that's really odd! I've seen people get a gigabit plus on these.

What phone do you have? Lots of times phones get new chips earlier than we can get them; it's very possible that your phone already supports NR aggregation of 3 or 4 bands. Phones also have highly tuned antennas; it's strange to see them beat some of our setups! If you share some more details maybe we can help improve it!

**Comment by qy2009** [comment]
I am on RM520N-GL, firmware AT+QGMR
RM520NGLAAR01A07M4G_01.001.01.001

When I was using Rooter + WG1608 from an external antenna, I got CSQ 31, RSSI -51, RSCP(dBm) RSRP -72 -69, ECIO(dB) RSRQ -11 , SINR 28.

I have iPhone 15 Pro Max. 

I have enabled ADB and set up TTL Mod(thank you for that :)) . 

Let me know what logs I can get to show you more details. 

thanks!

**Comment by suntecpro101** [comment]
I had a similar issue with my RM520 after update to RM520NGLAAR01A07M4G_01.201.01.201 problem solved

**Comment by qy2009** [comment]
thanks @suntecpro101 , would you mind sending the firmware to me? I tried messaging them on the forum for firmware, but haven't got a reply yet.

**Comment by suntecpro101** [comment]
https://quectel123-my.sharepoint.cn/:u:/g/personal/ae-fae_dom_quectel_com_cn/ET3WvUfnbvFFvgRr7uIF6mkBbYWUFlbVbnuhOR1atSN7cw?e=6vzgj4

**Comment by qy2009** [comment]
thanks so much, @suntecpro101 ! I was able to upgraded it to that version. it took me a while: 

1. the COM has to set speed to 115200;
2. run Qflash as admin, and also set the speed to be 115200;
3. choose the elf file under firehose folder;
4. choose AT port to do the upgrade.

I was stuck at failed to download firehose for a few times. and finally I was able to do it.

The forum support also sent me an updated firmware with new function to check 5g MIMO info :AT+QNWCFG="nr5g_mimo_info". however, I failed to upgrade to that newer version, same failed to download firehose message again, and I just gave up and keep the version you gave me :) 

Here is the link if you are interested in trying that version 💯 
https://cnquectel-my.sharepoint.com/:f:/g/personal/ae-fae_quectel_com/EiQVpDua8dlAq7QpNY4PnoMBz0ErEb4tsYm7aJSNgFtCKQ?e=BJRCib

now I am able to around 400Mbps for download, and upload around 100Mbps .
